workspaces/controller/api/v1beta1/workspacekind_types.go

```diff
@@ -62,3 +74,288 @@ type WorkspaceKindList struct {
+	Options KindOptions `json:"options"`
+}
+
+type WorkspaceKindIcon struct {
+	// TODO
+
+	//https://github.com/kubernetes-sigs/controller-tools/issues/461
+	//https://kubernetes.io/blog/2022/09/23/crd-validation-rules-beta/
+	//https://github.com/google/cel-spec/blob/master/doc/intro.md
+
+	// +kubebuilder:validation:XValidation:rule="has(self.url) && self.type == 'RequiredMember' ?  has(self.requiredMember) : !has(self.requiredMember)",message="requiredMember is required when type is RequiredMember, and forbidden otherwise"
+	// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'OptionalMember' ?  true : !has(self.optionalMember)",message="optionalMember is forbidden when type is not OptionalMember"
+
+	//+kubebuilder:example="https://jupyter.org/assets/favicons/apple-touch-icon-152x152.png"
+	Url string `json:"url,omitempty"`
+
+	ConfigMap WorkspaceKindConfigMap `json:"configMap,omitempty"`
+}
+
+type WorkspaceKindConfigMap struct {
+	//+kubebuilder:example="my-logos"
+	Name string `json:"name,omitempty"`
+	//+kubebuilder:example="apple-touch-icon-152x152.png"
+	Key string `json:"key,omitempty"`
+}
+
+// metadata for Workspace Pods (MUTABLE)
+type WorkspaceKindPodMetadata struct {
+	Labels      map[string]string `json:"labels,omitempty"`
+	Annotations map[string]string `json:"annotations,omitempty"`
+}
+
+type ServiceAccount struct {
+	// the name of the ServiceAccount; this Service Account MUST already exist in the Namespace of the Workspace,
+	//  the controller will NOT create it. We will not show this WorkspaceKind in the Spawner UI if the SA does not exist in the Namespace.
+	//+kubebuilder:example="default-editor"
+	Name string `json:"name"`
+}
+
+type WorkspaceKindCullingConfig struct {
+	// if the culling feature is enabled
+	//+kubebuilder:default=true
+	Enabled bool `json:"enabled,omitempty"`
+
+	// the maximum number of seconds a Workspace can be inactive
+	//+kubebuilder:default=86400
+	MaxInactiveSeconds int `json:"maxInactiveSeconds,omitempty"`
+
+	// the probe used to determine if the Workspace is active
+	ActivityProbe WorkspaceKindActivityProbe `json:"activityProbe,omitempty"`
+}
+
+type WorkspaceKindActivityProbe struct {
+	// a "shell" command to run in the Workspace, if the Workspace had activity in the last 60 seconds this command
+	//  should return status 0, otherwise it should return status 1
+	Exec WorkspaceKindActivityProbeExec `json:"exec,omitempty"`
+
+	// a Jupyter-specific probe will poll the /api/status endpoint of the Jupyter API, and use the last_activity field
+	//  Users need to be careful that their other probes don't trigger a "last_activity" update,
+	//	e.g. they should only check the health of Jupyter using the /api/status endpoint
+	Jupyter WorkspaceKindActivityProbeJupyter `json:"jupyter,omitempty"`
+}
+
+type WorkspaceKindActivityProbeExec struct {
+	// +TODO kubebuilder:example=["bash", "-c", "exit 0"]
```

Comment:
We can put full examples elsewhere, they don't need to be in the CRD spec, but this explains at least a little to users:

```suggestion
	// +kubebuilder:example=["bash", "-c", "exit 0"]
```